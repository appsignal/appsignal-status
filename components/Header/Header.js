import Link from "next/link";
import PropTypes from "prop-types";

const Header = ({ title }) => {
  return (
    <header className="bg-white shadow-sm">
      <div className="container flex justify-between items-center py-3">
        <nav className="flex space-x-2 sm:space-x-4">
          <a className="flex items-center no-underline font-semibold" href="/">
            {title}
          </a>
          <div className="h-6 w-px bg-gray-200 rounded" />
          <Link href="/">
            <a className="no-underline text-gray-700">Status</a>
          </Link>
        </nav>
      </div>
    </header>
  );
};

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
