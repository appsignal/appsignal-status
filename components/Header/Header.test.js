import React from "react";
import { render, screen } from "@testing-library/react";

import Header from "./Header";

describe("Header", () => {
  test("renders without errors", () => {
    const { container } = render(<Header title="Some title" />);
    expect(container).toMatchSnapshot();
  });

  test("renders the title", () => {
    render(<Header title="Dunder Mifflin" />);
    expect(screen.getByText("Dunder Mifflin")).toBeInTheDocument();
  });
});
