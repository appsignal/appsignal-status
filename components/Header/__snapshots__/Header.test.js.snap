// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without errors 1`] = `
<div>
  <header
    class="bg-white shadow-sm"
  >
    <div
      class="container flex justify-between items-center py-3"
    >
      <nav
        class="flex space-x-2 sm:space-x-4"
      >
        <a
          class="flex items-center no-underline font-semibold"
          href="/"
        >
          Some title
        </a>
        <div
          class="h-6 w-px bg-gray-200 rounded"
        />
        <a
          class="no-underline text-gray-700"
          href="/"
        >
          Status
        </a>
      </nav>
    </div>
  </header>
</div>
`;
